File: 232/implement-queue-using-stacks.java
class MyQueue {
    Stack<Integer> stack;
    Stack<Integer> temp;
    public MyQueue() {
        stack = new Stack<Integer>(); //real
        temp = new Stack<Integer>(); //temp
    }
    
    public void push(int x) {
        while(stack.size()>0){
            int now = stack.pop();
            temp.push(now);
        }
        stack.push(x);
        while(temp.size()>0){
            int now = temp.pop();
            stack.push(now);
        }
    }
    
    public int pop() {
        return stack.pop();
    }
    
    public int peek() {
        int now = stack.pop();
        stack.push(now);
        return now;
    }
    
    public boolean empty() {
        if(stack.size()==0) return true;
        else return false;
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */

File: 232/implement-queue-using-stacks.cpp
class MyQueue {
public:
    stack<int> q;
    stack<int> temp;
    MyQueue() {
        
    }
    
    void push(int x) {
        while(q.size()>0){
            temp.push(q.top());
            q.pop();
        }
        q.push(x);
        while(temp.size()>0){
            q.push(temp.top());
            temp.pop();
        }
    }
    
    int pop() {
        int temp = q.top();
        q.pop();
        return temp;
    }
    
    int peek() {
        return q.top();
    }
    
    bool empty() {
        return q.size()==0;
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */
